### PR TITLE
[RSS] Fix parsing logic

### LINKED ISSUE #24229: RSS Ticker fails on builds after v21.0a2-Omega
### Describe the bug

RSS Ticker no longer displays along bottom of Kodi window.

## Expected Behavior

Scrolling display of items from RSS feed sources.

## Actual Behavior

According to debug log feed URLs are seemingly fetched by Kodi, but fail to display silently with no further log entries until the idle connection is closed after a while.

## Possible Fix

Issue seems likely to relate to commit for migration of RSS Ticker functions to tinyxml2:-
[https://github.com/xbmc/xbmc/commit/fe6a7b1ed0de937b0b39c7c756dec0ab8c04f6a6](url)

Parent commit for TinyXML2 introduction:-
[https://github.com/xbmc/xbmc/commit/9e983ed0449aaf3007d8bfa3261056b120d84188](url)

Commit 9e983ed mentions that "Currently tinyxml2 and 1 can coexist peacefully" - perhaps revert the RSS Ticker files to use tinyxml1 ?

### To Reproduce

1.  Install a Kodi build later than v21.0a2-Omega
2.  In Settings/Interface activate "Show RSS news feeds"
3.  Have a set of valid RSS feeds in Kodi/userdata/RssFeeds.xml
 
### Debuglog

The debuglog can be found here: [kodi.log debuglog](https://gist.github.com/kuraikaze/f93048a5f2f5c7ab4ae5317781274325)

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [*] Linux
 - [ ] macOS
 - [*] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: 21.0-beta2

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by neo1973**
## Description
Fix logic error introduced in fe6a7b1ed0de937b0b39c7c756dec0ab8c04f6a6 and remove workaround that doesn't seem to be required with TinyXML2.

## Motivation and context
Fixes #24229.

## How has this been tested?
RSS feed is shown in _Arctic: Zephyr - Reloaded_ skin.

## What is the effect on users?
RSS works again.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

**Review by fuzzard — APPROVED** [review]
Maybe a test wouldnt go astray ;) otherwise my bad, and merge whenever

File: xbmc/utils/RssReader.cpp [hunk 1]
```diff
@@ -321,7 +321,7 @@ bool CRssReader::Parse(int iFeed)
     return false;
 
   std::string strValue = rootXmlNode->Value();
-  if (strValue.find("rss") == std::string::npos || strValue.find("rdf") == std::string::npos)
+  if (!(strValue.find("rss") == std::string::npos || strValue.find("rdf") == std::string::npos))
```
Comment (popcornmix) [inline xbmc/utils/RssReader.cpp:308]: The code used to be the else case of:
```
  if (strValue.find("rss") != std::string::npos ||
      strValue.find("rdf") != std::string::npos)
```
which if you want to invert (as the else case if being dropped) is:
```
  if (strValue.find("rss") == std::string::npos && strValue.find("rdf") == std::string::npos)
```
so I'm not sure this change is correct.

Comment (neo1973) [reply xbmc/utils/RssReader.cpp:308]: Right, no idea how I could fuck that one up :tired_face:

**Review by popcornmix — APPROVED** [review]
Looks fine now.

File: xbmc/utils/RssReader.cpp (hunk 1 above)
Comment (garbear) [reply xbmc/utils/RssReader.cpp]: De Morgan shall forgive you some day :wink:

**Comment by neo1973** [comment]
> Maybe a test wouldnt go astray

Added a very primitive one, better than nothing I guess :wink: Will merge sometime next week if no one complains in the meantime.

**Pull request merged**